File: week2/Pset2/readability/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

int determine_grade(string text);

int main(void)
{
    string text = get_string("Text: ");
    int grade = determine_grade(text);
    if (grade < 1)
    {
        printf("Before Grade 1\n");
        return 0;
    }
    if (grade > 16)
    {
        printf("Grade 16+\n");
        return 0;
    }
    printf("Grade %i\n", grade);
    return 0;
}

int determine_grade(string text)
{
    float words = 1, letters = 0, sentences = 0;
    for (int i = 0; i < strlen(text); i++)
    {
        if (isalpha(text[i]))
        {
            letters++;
        }
        else if (isblank(text[i]))
        {
            words++;
        }
        else if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            sentences++;
        }
    }
    return round(0.0588 * (letters / words * 100) - 0.296 * (sentences / words * 100) - 15.8);
}

File: week6/sentimental-readability/readability.py
#!/usr/local/bin/python3

from cs50 import get_string


def main():
    text = get_string("Text: ")
    grade = calc_grade(text)

    if grade < 1:
        print("Before Grade 1")
    elif grade >= 16:
        print("Grade 16+")
    else:
        print(f"Grade {round(grade)}")


def calc_grade(text):
    letters = 0
    for letter in text:
        if letter.isalpha():
            letters += 1

    words = len(text.split())
    sentences = text.count('.') + text.count('!') + text.count('?')

    L = letters / words * 100
    S = sentences / words * 100

    return 0.0588 * L - 0.296 * S - 15.8


main()
